src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
+                    {
+                        assm = Assembly.Load(assemblyName);
+                    }
+                    catch { }
+
+                    if (assm == null)
+                    {
+                        try
+                        {
+                            AssemblyName asmName = new AssemblyName(assemblyName);
+                            assm = Assembly.Load(asmName.Name);
+                        }
+                        catch { }
+                    }
+                }
+                else
+                {
+                    try
+                    {
+                        assm = Assembly.Load(new AssemblyName(assemblyName));
+                    }
+                    catch { }
```

Comment:
> In this code, whatever the value of bSimplAssembly, if either load succeeds it calls GetSimplyNamedTypeFromAssembly otherwise it calls FormatterServices.GetTypeFromAssembly.

No. If load succeeds and _isSimpleAssembly == true then it calls GetSimplyNamedTypeFromAssembly. If load succeeds and _isSimpleAssembly == false it calls  FormatterServices.GetTypeFromAssembly. If load fails it calls nothing.